tools/src/main/java/org/apache/kafka/tools/VerifiableProducer.java

```diff
@@ -242,66 +207,174 @@ public String getValue(long val) {
     /** Close the producer to flush any remaining messages. */
     public void close() {
         producer.close();
-        System.out.println(shutdownString());
+        printJson(new ShutdownComplete());
     }
 
-    String shutdownString() {
-        Map<String, Object> data = new HashMap<>();
-        data.put("name", "shutdown_complete");
-        return toJsonString(data);
+    @JsonPropertyOrder({ "timestamp", "name" })
```

Comment:
Is this way more explicit than leveraging on the order of the `put` execution ?
Maybe there are a lot of lines of code because other than doing this I have updated the verifiable producer to be quite similar to the verifiable consumer. See how I changed the `createFromArgs()` method and adding `ProducerEvent` class with all derived classes for defining event to log (like already happens with the verifiable consumer). Before these changes this tools seemed to be written by two completely different persons (maybe it's true ?) without considering that they do something similar (even if in the opposite direction). :-)